### PR TITLE
Add support for all action group  receiver types

### LINKED ISSUE #3603: Add missing features for Monitor Action Group senders
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Have a list of asks for the Azure Monitor Action Group resource:

1. When configuring a new Email/SMS/Push/Voice action within an Action Group via the portal, it is possible to enable the all of the different sender types (Email/SMS/Push/Voice) in a single action but when using the Terraform resource, you have to create a separate action for each type. Would be nice to be able to duplicate the functionality available in the portal

2. When creating an email receiver via Terraform, this creates a new action with the Email sender enabled and the email address. After attaching the action group to an Alert, we found that the even though the Alert was firing, it was not sending out emails via the Action Group. Once we went into the action via the portal and set `Enable the common alert schema` to `Yes`, emails would get sent each time the associated Alert fired.

3. Please add the ability to enable `Azure app Push Notifications` to the resource.

4. Please add the ability to enable the common alert schema to the resource.

### New or Affected Resource(s)

* azurerm_monitor_action_group

### Potential Terraform Configuration

```hcl
resource "azurerm_monitor_action_group" "secteam" {
  name                = "SecurityTeam"
  resource_group_name = "${data.terraform_remote_state.corerg.rgname}"
  short_name          = "SecTeam"

  email_receiver {
    name          = "${var.adm1_alertAction_name}_email"
    email_address = "${var.adm1_email}"
  }

  email_receiver {
    name          = "${var.adm2_alertAction_name}_email"
    email_address = "${var.adm2_email}"
  }

  email_receiver {
    name          = "${var.adm3_alertAction_name}_email"
    email_address = "${var.adm3_email}"
  }
}
```

### LINKED ISSUE #4203: Support all action group receivers
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I am interested in adding support for the other seven receiver types listed in the SDK in addition to the currently supported SMS, email, and webhook:

https://github.com/Azure/azure-sdk-for-go/blob/7a9d2769e4a581b0b1bc609c71b59af043e05c98/services/preview/monitor/mgmt/2019-03-01/insights/models.go#L682-L707 

### New or Affected Resource(s)

* azurerm_monitor_action_group

### Potential Terraform Configuration

```hcl
resource "azurerm_monitor_action_group" "test" {
  name                = "CriticalAlertsAction"
  resource_group_name = "${azurerm_resource_group.test.name}"
  short_name          = "p0action"

  itsm_receiver {
    name          = "createorupdateticket"
    workspace_id = "6eee3a18-aac3-40e4-b98e-1f309f329816"
    connection_id = "53de6956-42b4-41ba-be3c-b154cdf17b13"
    ticket_configuration = "{}"
    region = "southcentralus"
  }
  azure_app_push_receiver {
    name          = "pushtoadmin"
    email_address = "admin@contoso.com"
  }
  arm_role_receiver {
    "name": "Monitoring Reader",
    "roleId": "43d0d8ad-25c7-4714-9337-8ba259a9fe05",
    "useCommonAlertSchema": false
  }
  automation_runbook_receiver {}
  voice_receiver {}
  logic_app_receiver {}
  azure_function_receiver {} 
}
```

### PR DESCRIPTION
**Pull request by mcdafydd**
Related issue: #4203 

This PR adds support for the 6 additional action group receivers available in 

github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-03-01/insights

If acceptable, I would also suggest we update the insights API import paths to at least 

github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/2018-09-01/insights

which includes the arm_role_receiver type as well.  The Azure Portal now has a "secured webhook" type as well, but I don't see that in any of the API versions in azure-sdk-for-go master yet.

Fixes #4203

Edit: Removed "Fixes #3603".  Will submit fixes for that issue in a follow-on PR.

### CODE REVIEW

File: azurerm/data_source_monitor_action_group_test.go [hunk 1]
```diff
@@ -170,7 +220,41 @@ resource "azurerm_monitor_action_group" "test" {
   webhook_receiver {
     name        = "callmybackupapi"
     service_uri = "https://backup.example.com/warning"
+	}
+
+	automation_runbook_receiver {
+		name = "action_name_1"
+		automation_account_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-runbooks/providers/microsoft.automation/automationaccounts/aaa001"
```
Comment (tombuildsstuff) [inline azurerm/data_source_monitor_action_group_test.go]: could we provision [an automation account](https://www.terraform.io/docs/providers/azurerm/r/automation_account.html) dynamically too and reference it's ID here?

File: azurerm/data_source_monitor_action_group_test.go [hunk 2]
```diff
@@ -170,7 +220,41 @@ resource "azurerm_monitor_action_group" "test" {
...
+
+  voice_receiver {
+    name         = "remotesupport"
+    country_code = "86"
+    phone_number = "13888888888"
+	}
+
+	logic_app_receiver {
+		name = "logicappaction"
+		resource_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-logicapp/providers/Microsoft.Logic/workflows/logicapp"
```
Comment (tombuildsstuff) [inline azurerm/data_source_monitor_action_group_test.go]: could we provision [a logic app workflow](https://www.terraform.io/docs/providers/azurerm/r/logic_app_workflow.html) dynamically too and reference it's ID here?

File: azurerm/data_source_monitor_action_group_test.go [hunk 3]
```diff
@@ -170,7 +220,41 @@ resource "azurerm_monitor_action_group" "test" {
...
+		callback_url = "http://test-host:100/workflows/fb9c8d79b15f41ce9b12861862f43546/versions/08587100027316071865/triggers/manualTrigger/paths/invoke?api-version=2015-08-01-preview&sp=%%2Fversions%%2F08587100027316071865%%2Ftriggers%%2FmanualTrigger%%2Frun&sv=1.0&sig=IxEQ_ygZf6WNEQCbjV0Vs6p6Y4DyNEJVAa86U5B4xhk"
+	}
+
+	azure_function_receiver {
+		name = "funcaction"
+		function_app_resource_id = ""
```
Comment (tombuildsstuff) [inline azurerm/data_source_monitor_action_group_test.go]: could we provision [a function app](https://www.terraform.io/docs/providers/azurerm/r/function_app.html) dynamically too and reference it's ID here?

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 4]
```diff
@@ -146,9 +240,167 @@ func TestAccAzureRMMonitorActionGroup_webhookReceiver(t *testing.T) {
...
+					resource.TestCheckResourceAttr(resourceName, "sms_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "webhook_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "automation_runbook_receiver.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "automation_runbook_receiver.automation_account_id", "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-runbooks/providers/microsoft.automation/automationaccounts/aaa001"),
+					resource.TestCheckResourceAttr(resourceName, "automation_runbook_receiver.runbook_name", "my runbook"),
+					resource.TestCheckResourceAttr(resourceName, "automation_runbook_receiver.webhook_resource_id", "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-runbooks/providers/microsoft.automation/automationaccounts/aaa001/webhooks/webhook_alert"),
+					resource.TestCheckResourceAttrSet(resourceName, "automation_runbook_receiver.is_global_runbook"),
+					resource.TestCheckResourceAttr(resourceName, "automation_runbook_receiver.service_uri", "https://s13events.azure-automation.net/webhooks?token=randomtoken"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: heads up the `Import` step below will check these values match the ones in the config - as such we shouldn't need to compare this here :)

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 5]
```diff
@@ -440,38 +930,186 @@ resource "azurerm_monitor_action_group" "test" {
...
+}
+
+resource "azurerm_monitor_action_group" "test" {
+  name                = "acctestActionGroup-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+	short_name          = "acctestag"
+
+	logic_app_receiver {
+		name = "logicappaction"
+		resource_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-logicapp/providers/Microsoft.Logic/workflows/logicapp"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: (same here) could we dynamically provision a logic app workflow and use it's ID here?

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 6]
```diff
@@ -440,38 +930,186 @@ resource "azurerm_monitor_action_group" "test" {
...
+
+		voice_receiver {
+			name         = "remotesupport"
+			country_code = "86"
+			phone_number = "13888888888"
+		}
+
+		logic_app_receiver {
+			name = "logicappaction"
+			resource_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-logicapp/providers/Microsoft.Logic/workflows/logicapp"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: could we provision a Logic App and use it's id here instead?

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 7]
```diff
@@ -440,38 +930,186 @@ resource "azurerm_monitor_action_group" "test" {
...
+		}
+
+		webhook_receiver {
+			name        = "callmybackupapi"
+			service_uri = "https://backup.example.com/warning"
+		}
+
+		automation_runbook_receiver {
+			name = "action_name_1"
+			automation_account_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg-runbooks/providers/microsoft.automation/automationaccounts/aaa001"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: could we provision an Automation App and use it's id here instead?

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 8]
```diff
@@ -440,38 +930,186 @@ resource "azurerm_monitor_action_group" "test" {
...
+		logic_app_receiver {
+			name = "logicappaction"
+			resource_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-logicapp/providers/Microsoft.Logic/workflows/logicapp"
+			callback_url = "http://test-host:100/workflows/fb9c8d79b15f41ce9b12861862f43546/versions/08587100027316071865/triggers/manualTrigger/paths/invoke?api-version=2015-08-01-preview&sp=%%2Fversions%%2F08587100027316071865%%2Ftriggers%%2FmanualTrigger%%2Frun&sv=1.0&sig=IxEQ_ygZf6WNEQCbjV0Vs6p6Y4DyNEJVAa86U5B4xhk"
+			use_common_alert_schema = false
+		}
+
+		azure_function_receiver {
+			name = "funcaction"
+			function_app_resource_id = ""
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: could we provision a Function App and use it's id here instead?

File: go.mod [hunk 9]
```diff
@@ -13,10 +14,9 @@ require (
 	github.com/hashicorp/go-multierror v1.0.0
 	github.com/hashicorp/go-uuid v1.0.1
 	github.com/hashicorp/go-version v1.1.0
-	github.com/hashicorp/terraform v0.12.6
+	github.com/hashicorp/terraform v0.12.8
 	github.com/satori/go.uuid v1.2.0
 	github.com/satori/uuid v0.0.0-20160927100844-b061729afc07
-	github.com/sirupsen/logrus v1.2.0 // indirect
```
Comment (tombuildsstuff) [inline go.mod]: since there's no changes to the vendored files, could we revert the changes to the `go.mod` and `go.sum`?

File: azurerm/resource_arm_monitor_action_group.go [hunk 10]
```diff
@@ -66,6 +67,60 @@ func resourceArmMonitorActionGroup() *schema.Resource {
 				},
 			},
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group.go]: could we add all of these new blocks [to the documentation](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/r/monitor_action_group.html.markdown#L59)?

File: azurerm/data_source_monitor_action_group.go [hunk 11]
```diff
@@ -22,13 +24,15 @@ func dataSourceArmMonitorActionGroup() *schema.Resource {
 			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
 
 			"short_name": {
```
Comment (tombuildsstuff) [inline azurerm/data_source_monitor_action_group.go]: could we add all of these new blocks [to the documentation](could we add all of these new blocks [to the documentation](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/d/monitor_action_group.html.markdown#L42)?

File: go.mod (hunk 9 above)
Comment (mcdafydd) [reply go.mod]: Yes, thanks!  I had some module trouble initially trying to build the provider locally.

**Comment by mcdafydd** [comment]
> As this PR is being added I noticed that it still doesn't seem to have support for the UseCommonAlertSchema attribute which is exposed in the Go SDK and included in the example from #4203
> 
> Is that on purpose?

I wrote the issue initially with that in the proposed configuration before realizing that the ArmRoleReceiver doesn't appear until the azure-sdk-for-go 2018-09-01/insights version, and broad support for UseCommonAlertSchema appears in 2019-03-01/insights.  I ended up removing them both from the current commits and leaving the import version where it is at 2018-03-01.

I would ask for guidance about whether we can put these features back in and reference a newer API import in this PR, or if we should add them to a separate one.

Thanks!

File: azurerm/resource_arm_monitor_action_group.go (hunk 10 above)
Comment (mcdafydd) [reply azurerm/resource_arm_monitor_action_group.go]: I think we're all set here.

File: azurerm/data_source_monitor_action_group_test.go [hunk 12]
```diff
@@ -171,6 +221,114 @@ resource "azurerm_monitor_action_group" "test" {
...
+		"hello": {
+			"type": "string"
+		}
+	}
+}
+SCHEMA
+}
+
+resource "azurerm_storage_account" "test" {
+	name                     = "functionsapptestsa"
```
Comment (katbyte) [inline azurerm/data_source_monitor_action_group_test.go]: This is going to cause problems when running tests in parallel, could we change it to something random? maybe the last x digits of rInt?

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 13]
```diff
@@ -440,38 +930,328 @@ resource "azurerm_monitor_action_group" "test" {
...
+		service_uri = "https://s13events.azure-automation.net/webhooks?token=randomtoken"
+	}
+}
 
-  email_receiver {
-    name          = "sendtodevops"
-    email_address = "devops@contoso.com"
-  }
+resource "azurerm_automation_account" "test" {
+	name                = "aaa001"
```
Comment (katbyte) [inline azurerm/resource_arm_monitor_action_group_test.go]: This is going to cause issues with running tests in parallel, could we change it to be random?
```suggestion
	name                = "acctestAA-%d"
```

File: azurerm/data_source_monitor_action_group_test.go [hunk 14]
```diff
@@ -171,6 +221,114 @@ resource "azurerm_monitor_action_group" "test" {
...
+	azure_function_receiver {
+		name = "funcaction"
+		function_app_resource_id = "${azurerm_function_app.test.id}"
+		function_name = "myfunc"
+		http_trigger_url = "https://example.com/trigger"
+	}
+}
+
+resource "azurerm_automation_account" "test" {
+	name                = "aaa001"
```
Comment (katbyte) [inline azurerm/data_source_monitor_action_group_test.go]: Could we make this random?
```suggestion
	name                = "acctestAA-%d"
```

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 15]
```diff
@@ -440,38 +930,328 @@ resource "azurerm_monitor_action_group" "test" {
...
+	azure_function_receiver {
+		name = "funcaction"
+		function_app_resource_id = "${azurerm_function_app.test.id}"
+		function_name = "myfunc"
+		http_trigger_url = "https://example.com/trigger"
+	}
+}
+
+resource "azurerm_storage_account" "test" {
+	name                     = "functionsapptestsa"
```
Comment (katbyte) [inline azurerm/resource_arm_monitor_action_group_test.go]: And here

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 16]
```diff
@@ -440,38 +930,328 @@ resource "azurerm_monitor_action_group" "test" {
...
+			"hello": {
+				"type": "string"
+			}
+		}
+	}
+	SCHEMA
+	}
+
+	resource "azurerm_storage_account" "test" {
+		name                     = "functionsapptestsa"
```
Comment (katbyte) [inline azurerm/resource_arm_monitor_action_group_test.go]: And here

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 17]
```diff
@@ -254,8 +537,51 @@ func TestAccAzureRMMonitorActionGroup_singleReceiverUpdate(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
 					resource.TestCheckResourceAttr(resourceName, "email_receiver.#", "1"),
 					resource.TestCheckResourceAttr(resourceName, "email_receiver.0.email_address", "admin@contoso.com"),
+					resource.TestCheckResourceAttr(resourceName, "itsm_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "azure_app_push_receiver.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "sms_receiver.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "webhook_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "automation_runbook_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "voice_receiver.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "logic_app_receiver.#", "0"),
```
Comment (katbyte) [inline azurerm/resource_arm_monitor_action_group_test.go]: This check failed:`Check failed: Check 11/13 error: azurerm_monitor_action_group.test: Attribute 'logic_app_receiver.resource_id' not found`

File: azurerm/resource_arm_monitor_action_group_test.go (hunk 4 above)
Comment (mcdafydd) [reply azurerm/resource_arm_monitor_action_group_test.go]: @tombuildsstuff Thank you so much for all the helpful support on my first Terraform provider PR.  I'm still tracing through how Import works in the terraform helpers but will attend to this soon.

File: azurerm/data_source_monitor_action_group.go [hunk 18]
```diff
@@ -22,13 +24,15 @@ func dataSourceArmMonitorActionGroup() *schema.Resource {
...
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:         schema.TypeString,
+				Required:     true,
```
[the remark this answers is not in the document]
Comment (mcdafydd) [reply azurerm/data_source_monitor_action_group.go]: > This is a breaking change, could we make it optional?

I think I have this one fixed in the latest commits to make it computed.  The only thing Required now is the action group's "name".

File: azurerm/resource_arm_monitor_action_group_test.go (hunk 13 above)
Comment (mcdafydd) [reply azurerm/resource_arm_monitor_action_group_test.go]: Thanks, @katbyte!  I followed that naming scheme for all entries.  The only one with Azure length limits are the storage accounts for the function app at 24 characters, which I shortened, hopefully correctly.  In the go playground example, I could divide an 18-digit int by 1e12 and fmt specify it as %6.0f, but 'go vet' complained about that and wanted rInt/1e12 as an int.

File: azurerm/resource_arm_monitor_action_group_test.go (hunk 17 above)
Comment (mcdafydd) [reply azurerm/resource_arm_monitor_action_group_test.go]: I still need to look into this one.

**Comment by mcdafydd** [comment]
@tombuildsstuff 

Hi Tom - It looks like I accidentally forgot to click Submit Review on one of the two reviews.  Is there anything else I'm supposed to do to make sure this PR is ready to be merged?  Thanks!

**Comment by mcdafydd** [comment]
@katbyte Thanks for the heads up.  I see the changes in master to terraform-plugin-sdk and am working on https://www.terraform.io/docs/extend/plugin-sdk.html#step-2-migrate now.

**Comment by mcdafydd** [comment]
@katbyte we should be back in business.

File: azurerm/data_source_monitor_action_group_test.go [hunk 19]
```diff
@@ -171,11 +229,119 @@ resource "azurerm_monitor_action_group" "test" {
...
+		"hello": {
+			"type": "string"
+		}
+	}
+}
+SCHEMA
+}
+
+resource "azurerm_storage_account" "test" {
+	name                     = "acctestSA-%d"
```
Comment (tombuildsstuff) [inline azurerm/data_source_monitor_action_group_test.go]: dashes aren't valid in storage account names unfortunately - so this'd need to become:

```suggestion
	name                     = "acctestsa%s"
```

(from below it's also worth making this a string)

File: azurerm/data_source_monitor_action_group_test.go [hunk 20]
```diff
@@ -171,11 +229,119 @@ resource "azurerm_monitor_action_group" "test" {
...
+	app_service_plan_id       = "${azurerm_app_service_plan.test.id}"
+	storage_connection_string = "${azurerm_storage_account.test.primary_connection_string}"
 }
 
 data "azurerm_monitor_action_group" "test" {
   resource_group_name = "${azurerm_resource_group.test.name}"
   name                = "${azurerm_monitor_action_group.test.name}"
 }
-`, rInt, location, rInt)
+`, rInt, location, rInt, rInt, rInt, rInt/1e12, rInt, rInt)
```
Comment (tombuildsstuff) [inline azurerm/data_source_monitor_action_group_test.go]: to be consistent with the other resources it'd make sense to pass in a short (e.g. 4/5 char) string here:

```suggestion
`, rInt, location, rInt, rInt, rInt, rString, rInt, rInt)
```

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 21]
```diff
@@ -440,41 +954,319 @@ resource "azurerm_monitor_action_group" "test" {
...
+	azure_function_receiver {
+		name = "funcaction"
+		function_app_resource_id = "${azurerm_function_app.test.id}"
+		function_name = "myfunc"
+		http_trigger_url = "https://example.com/trigger"
+	}
+}
+
+resource "azurerm_storage_account" "test" {
+	name                     = "acctestSA-%d"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: (as above) we need to make this:

```suggestion
	name                     = "acctestsa%s"
```

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 22]
```diff
@@ -440,41 +954,319 @@ resource "azurerm_monitor_action_group" "test" {
...
+}
+
+resource "azurerm_function_app" "test" {
+	name                      = "acctestFA-%d"
+	location                  = "${azurerm_resource_group.test.location}"
+	resource_group_name       = "${azurerm_resource_group.test.name}"
+	app_service_plan_id       = "${azurerm_app_service_plan.test.id}"
+	storage_connection_string = "${azurerm_storage_account.test.primary_connection_string}"
+}
+`, rInt, location, rInt, rInt/1e12, rInt, rInt)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: (as above) can we pass in a random string instead?

```suggestion
`, rInt, location, rInt, rString, rInt, rInt)
```

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 23]
```diff
@@ -440,41 +954,319 @@ resource "azurerm_monitor_action_group" "test" {
...
+			"hello": {
+				"type": "string"
+			}
+		}
+	}
+	SCHEMA
+	}
+
+	resource "azurerm_storage_account" "test" {
+		name                     = "acctestSA-%d"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: (as above) could we make this:

```suggestion
		name                     = "acctestsa%s"
```

File: azurerm/resource_arm_monitor_action_group_test.go [hunk 24]
```diff
@@ -440,41 +954,319 @@ resource "azurerm_monitor_action_group" "test" {
...
+	}
+
+	resource "azurerm_function_app" "test" {
+		name                      = "acctestFA-%d"
+		location                  = "${azurerm_resource_group.test.location}"
+		resource_group_name       = "${azurerm_resource_group.test.name}"
+		app_service_plan_id       = "${azurerm_app_service_plan.test.id}"
+		storage_connection_string = "${azurerm_storage_account.test.primary_connection_string}"
+	}
+`, rInt, location, rInt, rInt, rInt, rInt/1e12, rInt, rInt)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_action_group_test.go]: (as above) could we pass in a random 4-5 char string here instead?

```suggestion
`, rInt, location, rInt, rInt, rInt, rString, rInt, rInt)
```

**Comment by mcdafydd** [comment]
Awesome!  👍 Thanks a lot for helping me through my first PR here @tombuildsstuff @katbyte

**Comment by mcdafydd** [comment]
Most recent commit should resolve merge conflicts from #4483.

**Comment by mcdafydd** [comment]
Closing this pull request in favor of #4638.
